sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeTest.java

```diff
@@ -671,6 +683,79 @@ public void showMandatoryDownloadReadyDialogTest() {
 
+    @Test
+    public void tryResetWorkflowWhenApplicationEnterForegroundWhenChannelNull() {
+
+        /* Mock download state. */
+        mockStatic(DistributeUtils.class);
+        when(DistributeUtils.getStoredDownloadState()).thenReturn(DOWNLOAD_STATE_COMPLETED);
+        final ServiceCall call = mock(ServiceCall.class);
+        doAnswer(new Answer<ServiceCall>() {
+
+            @Override
+            public ServiceCall answer(InvocationOnMock invocationOnMock) {
+                ((ServiceCallback) invocationOnMock.getArguments()[4]).onCallSucceeded(new HttpResponse(200, ""));
+                return call;
+            }
+        }).when(mHttpClient).callAsync(anyString(), anyString(), eq(Collections.<String, String>emptyMap()), any(HttpClient.CallTemplate.class), any(ServiceCallback.class));
+
+
+        /* Only invoking onStarting(), and not invoking Distribute.getInstance().onStarted() callback so that Channel is null. */
+        Distribute.getInstance().onStarting(mAppCenterHandler);
+
+        /* Start activity. */
+        Distribute.getInstance().onActivityStarted(mActivity);
+        Distribute.getInstance().onApplicationEnterForeground();
+        Distribute.getInstance().onActivityResumed(mActivity);
+
+        /* Verify download is not checked after we reset workflow. */
+        verify(mHttpClient, never()).callAsync(anyString(), anyString(), eq(Collections.<String, String>emptyMap()), any(HttpClient.CallTemplate.class), any(ServiceCallback.class));
+    }
+
+    @Test
+    public void tryResetWorkflowWhenApplicationEnterForegroundWhenChannelNotNull() {
+
+        /* Prepare data. */
+        mockStatic(DistributeUtils.class);
+        when(DistributeUtils.getStoredDownloadState()).thenReturn(DOWNLOAD_STATE_COMPLETED);
+        final ServiceCall call = mock(ServiceCall.class);
+        doAnswer(new Answer<ServiceCall>() {
+
+            @Override
+            public ServiceCall answer(InvocationOnMock invocationOnMock) {
+                ((ServiceCallback) invocationOnMock.getArguments()[4]).onCallSucceeded(new HttpResponse(200, ""));
+                return call;
+            }
+        }).when(mHttpClient).callAsync(anyString(), anyString(), eq(Collections.<String, String>emptyMap()), any(HttpClient.CallTemplate.class), any(ServiceCallback.class));
+
+        /* Start distribute. */
+        start();
+
+        /* Start activity. */
+        Distribute.getInstance().onActivityStarted(mActivity);
+        Distribute.getInstance().onApplicationEnterForeground();
+        Distribute.getInstance().onActivityResumed(mActivity);
+
+        /* Verify download is checked after we reset workflow. */
+        verify(mHttpClient).callAsync(anyString(), anyString(), eq(Collections.<String, String>emptyMap()), any(HttpClient.CallTemplate.class), any(ServiceCallback.class));
+
+        /* Stop activity. */
+        Distribute.getInstance().onActivityPaused(mActivity);
+        Distribute.getInstance().onActivityStopped(mActivity);
```

Comment:
We don't use onStop since lifecyle is mocked and we use Foreground/Backround instead (we still use pause and resume explicitly).